### PR TITLE
Pepper-46: filtering borked

### PR DESCRIPTION
**Pull request by SimoneMaiwald**
changed query creation logic from 
```"bool" : {
          "must" : [
            {
              "nested" : {
                "query" : {
                  "bool" : {
                    "must_not" : [
                      {
                        "exists" : {
                          "field" : "dsm.medicalRecord.faxSent",
                          "boost" : 1.0
                        }
                      }
                    ],
                    "adjust_pure_negative" : true,
                    "boost" : 1.0
                  }
                },
                "path" : "dsm.medicalRecord",
                "ignore_unmapped" : false,
                "score_mode" : "avg",
                "boost" : 1.0
              }
            },
            {
              "nested" : {
                "query" : {
                  "match" : {
                    "dsm.medicalRecord.mrProblem" : {
                      "query" : false,
                      "operator" : "AND",
                      "prefix_length" : 0,
                      "max_expansions" : 50,
                      "fuzzy_transpositions" : true,
                      "lenient" : false,
                      "zero_terms_query" : "NONE",
                      "auto_generate_synonyms_phrase_query" : true,
                      "boost" : 1.0
                    }
                  }
                },
                "path" : "dsm.medicalRecord",
                "ignore_unmapped" : false,
                "score_mode" : "avg",
                "boost" : 1.0
              }
            },
            {
              "nested" : {
                "query" : {
                  "match" : {
                    "dsm.medicalRecord.unableObtain" : {
                      "query" : false,
                      "operator" : "AND",
                      "prefix_length" : 0,
                      "max_expansions" : 50,
                      "fuzzy_transpositions" : true,
                      "lenient" : false,
                      "zero_terms_query" : "NONE",
                      "auto_generate_synonyms_phrase_query" : true,
                      "boost" : 1.0
                    }
                  }
                },
                "path" : "dsm.medicalRecord",
                "ignore_unmapped" : false,
                "score_mode" : "avg",
                "boost" : 1.0
              }
            },
            {
              "nested" : {
                "query" : {
                  "match" : {
                    "dsm.medicalRecord.duplicate" : {
                      "query" : false,
                      "operator" : "AND",
                      "prefix_length" : 0,
                      "max_expansions" : 50,
                      "fuzzy_transpositions" : true,
                      "lenient" : false,
                      "zero_terms_query" : "NONE",
                      "auto_generate_synonyms_phrase_query" : true,
                      "boost" : 1.0
                    }
                  }
                },
                "path" : "dsm.medicalRecord",
                "ignore_unmapped" : false,
                "score_mode" : "avg",
                "boost" : 1.0
              }
            }
          ],
          "adjust_pure_negative" : true,
          "boost" : 1.0
        }
      },
``` 
to 
```{
  "bool" : {
    "must" : [
      {
        "nested" : {
          "query" : {
            "bool" : {
              "must" : [
                {
                  "bool" : {
                    "must_not" : [
                      {
                        "exists" : {
                          "field" : "dsm.medicalRecord.faxSent",
                          "boost" : 1.0
                        }
                      }
                    ],
                    "adjust_pure_negative" : true,
                    "boost" : 1.0
                  }
                },
                {
                  "match" : {
                    "dsm.medicalRecord.mrProblem" : {
                      "query" : false,
                      "operator" : "AND",
                      "prefix_length" : 0,
                      "max_expansions" : 50,
                      "fuzzy_transpositions" : true,
                      "lenient" : false,
                      "zero_terms_query" : "NONE",
                      "auto_generate_synonyms_phrase_query" : true,
                      "boost" : 1.0
                    }
                  }
                },
                {
                  "match" : {
                    "dsm.medicalRecord.unableObtain" : {
                      "query" : false,
                      "operator" : "AND",
                      "prefix_length" : 0,
                      "max_expansions" : 50,
                      "fuzzy_transpositions" : true,
                      "lenient" : false,
                      "zero_terms_query" : "NONE",
                      "auto_generate_synonyms_phrase_query" : true,
                      "boost" : 1.0
                    }
                  }
                },
                {
                  "match" : {
                    "dsm.medicalRecord.duplicate" : {
                      "query" : false,
                      "operator" : "AND",
                      "prefix_length" : 0,
                      "max_expansions" : 50,
                      "fuzzy_transpositions" : true,
                      "lenient" : false,
                      "zero_terms_query" : "NONE",
                      "auto_generate_synonyms_phrase_query" : true,
                      "boost" : 1.0
                    }
                  }
                }
              ],
              "adjust_pure_negative" : true,
              "boost" : 1.0
            }
          },
          "path" : "dsm.medicalRecord",
          "ignore_unmapped" : false,
          "score_mode" : "avg",
          "boost" : 1.0
        }
      },```

### CODE REVIEW

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/filter/query/BaseAbstractQueryBuilder.java [hunk 1]
```diff
@@ -90,7 +95,23 @@ protected void buildUpQuery(List<String> filterValues, FilterStrategy filterStra
...
                     QueryStrategyFactory.create(new QueryStrategyFactoryPayload(baseQueryBuilder, operator, parser)).build();
-            filterStrategy.build(boolQueryBuilder, baseQueryBuilder.build(queryBuilders));
+
+            if (StringUtils.isNotBlank(path)) {
+                if (!nestedQueryBuilderLists.containsKey(path)) {
+                    nestedQueryBuilderLists.put(path, queryBuilders);
+                    correctBaseQueryBuilder.put(path, baseQueryBuilder);
+                } else {
+                    List<QueryBuilder> queryBuildersFromMap = nestedQueryBuilderLists.get(path);
+                    queryBuildersFromMap.addAll(queryBuilders);
```
Comment (pegahtah) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/filter/query/BaseAbstractQueryBuilder.java:105]: I could be confused, but shouldn't you add that again to the hashmap?
```nestedQueryBuilderLists.put(path, queryBuildersFromMap);```

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/filter/query/BaseAbstractQueryBuilder.java [hunk 2]
```diff
@@ -90,7 +95,23 @@ protected void buildUpQuery(List<String> filterValues, FilterStrategy filterStra
...
+                }
+            } else {
+                filterStrategy.build(boolQueryBuilder, baseQueryBuilder.build(queryBuilders));
+            }
+        }
+        if (!nestedQueryBuilderLists.isEmpty() && nestedQueryBuilderLists.size() == correctBaseQueryBuilder.size()) {
```
Comment (pegahtah) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/filter/query/BaseAbstractQueryBuilder.java:111]: Can the size ever not match?

Comment (SimoneMaiwald) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/filter/query/BaseAbstractQueryBuilder.java:111]: not really, but I am still paranoid

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/filter/query/BaseAbstractQueryBuilder.java (hunk 1 above)
Comment (SimoneMaiwald) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/filter/query/BaseAbstractQueryBuilder.java:105]: nope

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/filter/query/BaseAbstractQueryBuilder.java (hunk 2 above)
Comment (pegahtah) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/filter/query/BaseAbstractQueryBuilder.java:111]: Maybe then we can throw an error in that case to know why this part was skipped?

**Review by pegahtah — APPROVED** [review]

**Review by ssettipalli — APPROVED** [review]

**Pull request merged**